x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/components/multi_text_input.tsx

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useCallback, useState, useEffect } from 'react';
+import type { FunctionComponent, ChangeEvent } from 'react';
+
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiButtonIcon,
+  EuiSpacer,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+interface Props {
+  value: string[];
+  onChange: (newValue: string[]) => void;
+  errors?: Array<{ message: string; index?: number }>;
+  isInvalid?: boolean;
+  isDisabled?: boolean;
+}
+
+interface RowProps {
+  index: number;
+  value: string;
+  onChange: (index: number, value: string) => void;
+  onDelete: (index: number) => void;
+  autoFocus?: boolean;
+  isDisabled?: boolean;
+  multi?: boolean;
+}
+
+const Row: FunctionComponent<RowProps> = React.memo(
+  ({ index, value, onChange, onDelete, autoFocus, isDisabled, multi }) => {
+    const onDeleteHandler = useCallback(() => {
+      onDelete(index);
+    }, [onDelete, index]);
+
+    const onChangeHandler = useCallback(
+      (e: ChangeEvent<HTMLInputElement>) => {
+        onChange(index, e.target.value);
+      },
+      [onChange, index]
+    );
+
+    return (
+      <EuiFlexGroup alignItems="center" gutterSize="none" responsive={false}>
+        <EuiFlexItem>
+          <EuiFieldText
+            fullWidth
+            value={value}
+            onChange={onChangeHandler}
+            autoFocus={autoFocus}
+            disabled={isDisabled}
+          />
+        </EuiFlexItem>
+        {multi && (
+          <EuiFlexItem grow={false}>
+            <EuiButtonIcon
+              color="text"
+              onClick={onDeleteHandler}
+              iconType="cross"
+              disabled={isDisabled}
+              aria-label={i18n.translate('xpack.fleet.multiTextInput.deleteRowButton', {
+                defaultMessage: 'Delete row',
+              })}
+            />
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroup>
+    );
+  }
+);
+
+function defaultValue(value: string[]) {
+  return value.length > 0 ? value : [''];
+}
+
+export const MultiTextInput: FunctionComponent<Props> = ({
```

Comment:
any value to to memo this too?

Reply:
I do not think so actually I think there is probably no good reason to use for Row too, I will remove it.